packages/amplify_storage_s3/android/src/main/kotlin/com/amazonaws/amplify/amplify_storage_s3/types/FlutterDownloadFileRequest.kt

```diff
@@ -47,15 +48,14 @@ data class FlutterDownloadFileRequest(val request: Map<String, *>) {
     }
 
     companion object {
-        fun isValid(request: Map<String, *>): Boolean {
-            var valid = true
-            if (request["path"] !is String) {
-                valid = false
+        private const val validationErrorMessage: String = "Download request malformed."
+        fun isValid(request: Map<String, *>) {
+            if(request["path"] !is String?) {
+                throw AmplifyFlutterValidationException(validationErrorMessage, "path is missing.")
             }
-            if (request["key"] !is String) {
-                valid = false
+            if(request["key"] !is String?) {
+                throw AmplifyFlutterValidationException(validationErrorMessage, "key is missing.")
```

Comment:
Isn't the second argument a `recoverySuggestion`?

Reply:
Fixed.